lib/js/jslib.js

```diff
@@ -1757,11 +1757,30 @@ function _getInputValue(id) {
   var element = document.getElementById(id);
   DEBUG.assert(element != null, "invalid input node (id " + id + ")");
   DEBUG.assert(element.value != undefined, "invalid input value in id " + id)
-  return element.value;
+  if ((element.type != "radio" && element.type != "checkbox") || element.checked) {
+    return element.value;
+  } else {
+    return "";
+  }
 }
 
 var getInputValue = LINKS.kify(_getInputValue);
 
+// getRadioGroupValue : [String] -> String
+function _getRadioGroupValue(ids) {
+  let ptr = ids;
+  while ( LINKEDLIST.isNil(ptr) == false ) {
```

Comment:
Dare I say that we ought to use strict equality here `===`? Without having looked at the implementation of `isNil`, my faint hope is that it returns either `true` or `false` and not any other truthy or falsy value.

Reply:
Happy to fix this, but I based this on https://github.com/links-lang/links/blob/faa49cd09cc20429122c9def648b25cc3cab991b/lib/js/jslib.js#L254 

so maybe that should be changed to use `===` too.

Reply:
We should possibly make an issue to clean up `jslib.js`.

Reply:
Be my guest - I have only looked at enough of `jslib` to figure out how to solve this specific problem.